backend/compact-connect/lambdas/cloudfront-csp/index.js

```diff
@@ -1,5 +1,337 @@
+    }
+
+    return requestDomain;
+};
+
+/**
+ * Get a fully qualified domain URI with the protocol scheme.
+ * @param  {string} domain The bare domain string.
+ * @return {string}        The fully-qualified domain string.
+ */
+const getFullyQualified = (domain) => {
+    const protocol = 'https://';
+    let fullyQualified = '';
+
+    if (domain && typeof domain === 'string' && !domain.startsWith(protocol)) {
+        fullyQualified = `${protocol}${domain}`;
+    }
+
+    return fullyQualified;
+};
+
+/**
+ * Helper to get the fully-qualified domains for connected services based on the request domain.
+ * @param  {string} requestDomain The bare domain of the request.
+ * @return {object}               A map of fully-qualified domains for the request environment.
+ *   @return {string} dataApi       The data API fully-qualified domain.
+ *   @return {string} s3Upload      The S3 fully-qualified domain for uploading state files.
+ */
+const getEnvironmentUrls = (requestDomain) => {
+    const environmentUrls = {};
+    let environment;
+
+    switch (requestDomain) {
+    case environments.csg.prod.webFrontend:
+        environment = environments.csg.prod;
+        break;
+    case environments.csg.test.webFrontend:
+        environment = environments.csg.test;
+        break;
+    case environments.ia.prod.webFrontend:
+        environment = environments.ia.prod;
+        break;
+    case environments.ia.test.webFrontend:
+        environment = environments.ia.test;
+        break;
+    case environments.ia.justin.webFrontend:
+        environment = environments.ia.justin;
+        break;
+    default:
+        environment = environments.csg.prod;
+        break;
+    }
+
+    environmentUrls.dataApi = getFullyQualified(environment.dataApi);
+    environmentUrls.s3Upload = getFullyQualified(environment.s3Upload);
+
+    return environmentUrls;
+};
+
+/**
+ * Helper to escape CSP src keywords.
+ * @param  {string} keyword The standard keyword value.
+ * @return {string}         The escaped keyword value.
+ */
+const srcKeywordEscape = (keyword) => {
+    let escaped = '';
+
+    if (keyword && typeof keyword === 'string') {
+        escaped = `'${keyword}'`.toLowerCase();
+    }
+
+    return escaped;
+};
+
+/**
+ * Helper to automatically escape and prep CSP src keyword values (by reference).
+ * @param  {Array<string>} srcList       The CSP src list.
+ * @param  {string}        [listName=''] Optional src list name for logging.
+ */
+const srcKeywordsEscape = (srcList, listName = '') => {
+    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy#keyword_values
+    const srcKeywordsConfig = [
+        { value: 'self', isAllowed: true },
+        { value: 'none', isAllowed: true },
+        { value: 'strict-dynamic', isAllowed: true },
+        { value: 'report-sample', isAllowed: true },
+        { value: 'inline-speculation-rules', isAllowed: true },
+        { value: 'unsafe-inline', isAllowed: true },
```

Comment:
I see the word `unsafe`, and I get scared 😆 ... do we need to allow this? if yes, add a brief comment above?

Reply:
Good catch. We weren't actually using it, but it should be disallowed nonetheless.